crates/db_schema/src/schema_setup/mod.rs

```diff
@@ -0,0 +1,52 @@
+use anyhow::Context;
+use diesel::{connection::SimpleConnection, Connection, PgConnection};
+use diesel_migrations::{EmbeddedMigrations, MigrationHarness};
+use lemmy_utils::error::LemmyError;
+use std::path::Path;
+use tracing::info;
+
+pub const MIGRATIONS: EmbeddedMigrations = embed_migrations!();
+
+/// This SQL code sets up the `r` schema, which contains things that can be safely dropped and replaced
+/// instead of being changed using migrations. It may not create or modify things outside of the `r` schema
+/// (indicated by `r.` before the name), unless a comment says otherwise.
+///
+/// If you add something that depends on something (such as a table) created in a new migration, then down.sql
+/// must use `CASCADE` when dropping it. This doesn't need to be fixed in old migrations because the
+/// "replaceable-schema" migration runs `DROP SCHEMA IF EXISTS r CASCADE` in down.sql.
+const REPLACEABLE_SCHEMA: &[&str] = &[
+  "BEGIN;",
+  "DROP SCHEMA IF EXISTS r CASCADE;",
+  "CREATE SCHEMA r;",
+  include_str!("utils.sql"),
+  include_str!("triggers.sql"),
+  "COMMIT;",
+];
```

Comment:
Does this mean the schema is dropped and recreated every time Lemmy is started? That would be rather inefficient.

Reply:
Anything that's not in a migration needs to be recreated

Reply:
I spose its not too big a problem, especially if these run quickly and the startup is blocked before they complete.